### PR TITLE
Appropiate logging when a LockTimeout for VCS is reached

### PR DESCRIPTION
**Pull request by humitos**
Related #3856

### CODE REVIEW

File: readthedocs/projects/tasks.py [hunk 1]
```diff
@@ -218,10 +218,24 @@ def run(self, version_pk):  # pylint: disable=arguments-differ
         except RepositoryError:
             # Do not log as ERROR handled exceptions
             log.warning('There was an error with the repository', exc_info=True)
+        except vcs_support_utils.LockTimeout:
+            log.info(
+                'Lock still active: project=%s version=%s',
+                self.project.slug,
+                self.version.slug,
+            )
```
Comment (humitos) [inline readthedocs/projects/tasks.py]: This should raise `VersionLockedError` here.

File: readthedocs/projects/tasks.py [hunk 2]
```diff
@@ -548,7 +562,31 @@ def setup_vcs(self):
...
+            self.sync_repo()
+        except RepositoryError:
+            # Do not log as ERROR handled exceptions
+            log.warning('There was an error with the repository', exc_info=True)
+        except vcs_support_utils.LockTimeout:
+            log.info(
+                'Lock still active: project=%s version=%s',
+                self.project.slug,
+                self.version.slug,
+            )
```
Comment (humitos) [inline readthedocs/projects/tasks.py]: Same here.

**Comment by humitos** [comment]
Yes, I also noted those. I need to fix them.

**Comment by ericholscher** [comment]
Believe we can ignore the codecov issues here -- it feels wrong, and I will merge.

**Pull request merged**